pkg/tsdb/azuremonitor/azure-log-analytics-datasource.go

```diff
@@ -248,153 +276,29 @@ func (e *AzureLogAnalyticsDatasource) unmarshalResponse(res *http.Response) (Azu
-				}
-
-				if metricIndex == -1 && v.Type == "string" {
-					metricIndex = i
-				}
-
-				if valueIndex == -1 && (v.Type == "int" || v.Type == "long" || v.Type == "real" || v.Type == "double") {
-					valueIndex = i
-				}
-			}
-
-			if timeIndex == -1 {
-				azlog.Info("No time column specified. Returning existing columns, no data")
-				return nil, simplejson.NewFromAny(meta), nil
-			}
-
-			if valueIndex == -1 {
-				azlog.Info("No value column specified. Returning existing columns, no data")
-				return nil, simplejson.NewFromAny(meta), nil
-			}
-
-			slice := tsdb.TimeSeriesSlice{}
-			buckets := map[string]*tsdb.TimeSeriesPoints{}
-
-			getSeriesBucket := func(metricName string) *tsdb.TimeSeriesPoints {
-				if points, ok := buckets[metricName]; ok {
-					return points
-				}
-
-				series := tsdb.NewTimeSeries(metricName, []tsdb.TimePoint{})
-				slice = append(slice, series)
-				buckets[metricName] = &series.Points
-
-				return &series.Points
-			}
-
-			for _, r := range t.Rows {
-				timeStr, ok := r[timeIndex].(string)
-				if !ok {
-					return nil, simplejson.NewFromAny(meta), errors.New("invalid time value")
-				}
-				timeValue, err := time.Parse(time.RFC3339Nano, timeStr)
-				if err != nil {
-					return nil, simplejson.NewFromAny(meta), err
-				}
-
-				var value float64
-				if value, err = getFloat(r[valueIndex]); err != nil {
-					return nil, simplejson.NewFromAny(meta), err
-				}
-
-				var metricName string
-				if metricIndex == -1 {
-					metricName = t.Columns[valueIndex].Name
-				} else {
-					metricName, ok = r[metricIndex].(string)
-					if !ok {
-						return nil, simplejson.NewFromAny(meta), err
-					}
-				}
-
-				points := getSeriesBucket(metricName)
-				*points = append(*points, tsdb.NewTimePoint(null.FloatFrom(value), float64(timeValue.Unix()*1000)))
-			}
-
-			return slice, simplejson.NewFromAny(meta), nil
-		}
-	}
-
-	return nil, nil, errors.New("no data as no PrimaryResult table was returned in the response")
+	azlog.Error("failed to encode the query into the encodedQuery property")
 }
 
-func createMetadata(model *simplejson.Json, params url.Values) (metadata, error) {
-	meta := metadata{
-		Query:        params.Get("query"),
-		Subscription: model.Get("subscriptionId").MustString(),
-		Workspace:    model.Get("azureLogAnalytics").Get("workspace").MustString(),
-	}
-
-	encQuery, err := encodeQuery(meta.Query)
-	if err != nil {
-		return meta, err
-	}
-	meta.EncodedQuery = encQuery
-	return meta, nil
-}
-
-func encodeQuery(rawQuery string) (string, error) {
+// encodeQuery encodes the query in gzip so the frontend can build links.
+func encodeQuery(rawQuery string) ([]byte, error) {
 	var b bytes.Buffer
 	gz := gzip.NewWriter(&b)
 	if _, err := gz.Write([]byte(rawQuery)); err != nil {
-		return "", err
+		return nil, err
 	}
 
 	if err := gz.Close(); err != nil {
-		return "", err
+		return nil, err
 	}
 
-	return base64.StdEncoding.EncodeToString(b.Bytes()), nil
+	return b.Bytes(), nil
```

Comment:
Isn't `base64.StdEncoding.EncodeToString` needed any longer due to data frames? Have you verified deep linking works without this?

Reply:
I haven't verified deep linking yet, realized that after I marked the PR for review - and waiting for Ryan to make the changes on the frontend to get the metadata from the frame now. 

Then I will verify it :-)

I think it should work though, because I have made it []byte, and the default json encoding for []byte is to base64 encode it. Although perhaps because it is an interface with a concrete type of []byte, it won't work? We shall see.

Reply:
Ok :+1: Let me know if you want another review

Reply:
Looking the Inspect: JSON: DataFrame structure side by side with the old version, it appears it the encoded query is still base64 encoded.

Reply:
@marefr this is working now, tested the deeplinks and verified it takes me to the working query in the azure portal.